pkg/controller/servingcert/controller/secret_updating_controller.go

```diff
@@ -69,70 +72,142 @@ func (sc *serviceServingCertUpdateController) Key(namespace, name string) (metav
 
 func (sc *serviceServingCertUpdateController) Sync(obj metav1.Object) error {
 	sharedSecret := obj.(*v1.Secret)
+	service := sc.getServiceForSecret(sharedSecret)
+	if service == nil {
+		return nil
+	}
 
-	regenerate, service := sc.requiresRegeneration(sharedSecret)
-	if !regenerate {
+	if !isSecretValidForService(service, sharedSecret) {
 		return nil
 	}
 
 	// make a copy to avoid mutating cache state
 	secretCopy := sharedSecret.DeepCopy()
 
-	if err := toRequiredSecret(sc.dnsSuffix, sc.ca, service, secretCopy); err != nil {
+	update := false
+	if sc.requiresRegeneration(service, sharedSecret, sc.minTimeLeftForCert) {
+		// Regenerate the secret
+		if err := toRequiredSecret(sc.dnsSuffix, sc.ca, service, secretCopy); err != nil {
+			return err
+		}
+		_, err := sc.secretClient.Secrets(secretCopy.Namespace).Update(secretCopy)
 		return err
 	}
+	// If not regenerating, perform checks here to
+	// 1. ensure only the 2 data keys we want exist, TLSCertKey and TLSPrivateKeyKey
+	// 2. ensure cert data is at least a parseable certificate, if not replace with valid data
+	// This does not ensure that somebody didn't swap out secret data for another valid cert.
+	var err error
+	update, err = sc.ensureSecretData(service, secretCopy)
```

Comment:
Nit: you can scrape `update` and `err` definition and declaration above now and only have `update, err := sc.ensureSecretData(service, secretCopy)`

Reply:
yes, you are correct : ) done